src/main/java/ch/njol/skript/lang/function/FunctionReference.java

```diff
@@ -123,7 +123,9 @@ public boolean validateFunction(boolean first) {
 		function = null;
 		SkriptLogger.setNode(node);
 		Skript.debug("Validating function " + functionName);
-		Signature<?> sign = Functions.getSignature(functionName, script);
+		Signature<?> sign = Functions.getLocalSignature(functionName, script);
+		if (sign == null)
+			sign = Functions.getGlobalSignature(functionName);
```

Comment:
I think overloading the `Functions.getSignature` method would be better than checking if the signature exists every time. It would look something like `Functions.getSignature(String name, String script)`.

Same goes for functions ^

Reply:
That doesn't include the old methods, the whole point of this pull request. To not break compatibility.

Reply:
i never said to remove the old methods, i said to overload the methods since they don't have the same signature

Reply:
I don't think you know what overloading means.

Reply:
I'm pretty sure I do

Reply:
I never said to remove the different methods :shrug: I just said to combine them in a method where it searches for both, instead of checking for them every time